src/frontend/komponenter/Behandling/Brev/Brevmeny.tsx

```diff
@@ -51,39 +51,50 @@ interface Props {
     settKanSendesTilBeslutter: (kanSendesTilBeslutter: boolean) => void;
 }
 
-const Brevmeny: React.FC<Props> = ({
-    settBrevRessurs,
-    behandlingId,
-    personopplysninger,
-    settKanSendesTilBeslutter,
-}) => {
+const brevMal = 'innvilgetOvergangsstonadHoved2';
+const datasett = 'ef-brev';
+
+const initFlettefelterMedVerdi = (brevStruktur: BrevStruktur): FlettefeltMedVerdi[] =>
+    brevStruktur.flettefelter.flettefeltReferanse.map((felt) => ({
+        _ref: felt._id,
+        verdi: null,
+    }));
+
+const Brevmeny: React.FC<Props> = (props) => {
     const { axiosRequest } = useApp();
     const [brevStruktur, settBrevStruktur] = useState<Ressurs<BrevStruktur>>(byggTomRessurs());
-    const [alleFlettefelter, settAlleFlettefelter] = useState<FlettefeltMedVerdi[]>([]);
-    const [valgteFelt, settValgteFelt] = useState<ValgtFelt>({});
-    const [valgteDelmaler, settValgteDelmaler] = useState<{ [delmalNavn: string]: boolean }>({});
-
-    const brevMal = 'innvilgetOvergangsstonadHoved2';
-    const datasett = 'ef-brev';
 
     useEffect(() => {
         axiosRequest<BrevStruktur, null>({
             method: 'GET',
             url: `/familie-brev/api/${datasett}/avansert-dokument/bokmaal/${brevMal}/felter`,
         }).then((respons: Ressurs<BrevStruktur>) => {
-            if (respons.status === RessursStatus.SUKSESS) {
-                settAlleFlettefelter(
-                    respons.data.flettefelter.flettefeltReferanse.map((felt) => ({
-                        _ref: felt._id,
-                        verdi: null,
-                    }))
-                );
-                settBrevStruktur(respons);
-            }
+            settBrevStruktur(respons);
         });
         // eslint-disable-next-line
     }, []);
 
+    return (
+        <DataViewer response={{ brevStruktur }}>
+            {({ brevStruktur }) => <BrevmenyView {...props} brevStruktur={brevStruktur} />}
+        </DataViewer>
+    );
+};
+
+const BrevmenyView: React.FC<Props & { brevStruktur: BrevStruktur }> = ({
```

Comment:
`BrevmenyView`? 🙅

Reply:
Tja, skjønner jo hva det er, men det er jo litt "engelsk"?

Reply:
Det var bare ett navn for å splitte på Brevmeny med kallet som henter brevStruktur, og den som faktiskt renderer allt

Reply:
Ser den, men kan egentlig alt dette flyttes opp til `Brev.tsx` ? Dataviewer og uthenting av brevstruktur?